src/userbase-server/db.js

```diff
@@ -197,62 +199,35 @@ const rollbackTransaction = async function (transaction) {
 
 exports.rollbackTransaction = rollbackTransaction
 
-const failedTxConditionCheckMsg = 'Make sure user has write permission to this db and the db id and hash are correct'
 const putTransaction = async function (transaction, userId, dbNameHash, databaseId) {
-  const user = await userController.getUserByUserId(userId)
+  const [user, database] = await Promise.all([
+    userController.getUserByUserId(userId),
+    getDatabase(userId, dbNameHash)
```

Comment:
What if we don't check for this here, and instead put it in the websocket session where we have the list of open databases? Maybe we could poll each open database every 10 seconds, and update the local state with its metadata (isClosed, isReadOnly, etc). So during putTransaction we would only do just 1 DDB write (an insert into the transactions table). It would improve latency, and reduce the failure modes. The only downside seems to be that putting a DB in readOnly would take a few seconds to take effect, but I think that's a preferable tradeoff.

Reply:
This is doable and the same logic applies from my [other comment](https://github.com/encrypted-dev/userbase/pull/26#discussion_r354541246). Technically even with this approach in the PR, it's still possible to insert a transaction when readOnly is true (though the window of time it's possible is much smaller than the poll interval)

In my opinion, the polling approach seems like a relatively complex optimization to maintain and the benefit isn't super high (marginal latency improvement since the database queries use partition keys + the failure modes are still the same except for a DDB timeout which is pretty simple to handle). But, nonetheless, very doable and does make sense. Happy to do it

Reply:
Let's just remove the check for now, and not do the polling. The check is likely doubling our latency for little benefit. We can add the polling when we add the Admin API to set a DB to read-only or when DB sharing is finished.

Reply:
Can absolutely see that people would definitely want the lowest possible latency when inserting a transaction